server/aesdsocket.c

```diff
@@ -0,0 +1,352 @@
+//    if (sigaction(SIGCHLD, &signal_action, NULL) == -1) {
+//        perror("sigaction");
+//        exit(1);
+//    }
+    printf("Currently listening for connections!\n");
+
+    while (1) {  // main accept() loop
+        sin_size = sizeof their_addr;
+        client_socket = accept(socket_fd, (struct sockaddr *) &their_addr, &sin_size);
+        if (client_socket == -1) {
+            perror("accept");
+            continue;
+        }
+
+        inet_ntop(their_addr.ss_family,
+                  get_in_addr((struct sockaddr *) &their_addr),
+                  s, sizeof s);
+        printf("server: got connection from %s \r\n", s);
+        syslog(LOG_INFO, "Accepted connection from %s \r\n", s);
+
+
+        
+        bool end_of_packet = false;
+        uint32_t packet_buffer_pos = 0;
+
+
+        char buffer[BUFFER_SIZE*20];
+
+        ssize_t bytes_received;
+
+        //Allocate for the packet buffer
+        packet_buffer = (char *) malloc(BUFFER_SIZE*20);
+
+        while (1) {
+            bytes_received = recv(client_socket, buffer, BUFFER_SIZE, 0);
+            end_of_packet = false;
+            if (bytes_received == -1) {
+                perror("recv");
+                close(client_socket);
+                break;
+            } else if (bytes_received == 0) {
+                // Connection closed by the client
+                syslog(LOG_INFO, "Closed connection from %s \r\n", s);
+                printf("Connection closed \r\n");
+                close(client_socket);
+                break;
+            }
+            //Iterate through the bytes received and add them to the buffer to write to the file
+            for (int i = 0; i < bytes_received; i++) {
+                packet_buffer[packet_buffer_pos++] = buffer[i];
+                //Keep appending bytes till a newline is seen
+                if (buffer[i] == '\n') {
+                    packet_buffer[packet_buffer_pos] = '\0';
+                    packet_buffer_pos = 0;
+                    end_of_packet = true;
+                    break;
+                }
+
+            }
+
+            if (end_of_packet) {
+                //Echo the packet serverside just to make sure
+                printf("Incoming packet contained %s", packet_buffer);
+
+                //Create the output file if it doesn't exist
+                int log_fd = open(OUTPUT_FILE, O_WRONLY | O_CREAT | O_APPEND, 0666);
+
+                //Append the contents of the packet to the file
+                ssize_t bytes_written = write(log_fd, packet_buffer, strlen(packet_buffer));
+                if (bytes_written == -1) {
+                    perror("child process failed to write to file");
+					free(packet_buffer);
+                    close(log_fd);
+                    return 1; // Return an error code
+                }
+                close(log_fd);
+                // Check if a packet has finish being received and then send back the contents of the file
+                if (!fork()) {
```

Comment:
@Silex93 you should fork earlier and handle the actual received packet in the child process.